### PR TITLE
[Python3 migration]Fix test failure in qos_sai_base.py

### PR DESCRIPTION
**Pull request by wsycqyz**
### Description of PR

Summary:
Fixes test failure in qos_sai_base.py after Python3 migration

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Test cases fail in qos_sai_base.py after Python3 migration.
There are a lot of encode()/replace()/translate() in tests/qos. These functions behave differently in Python3.

#### How did you do it?
1. In Python 2, the encode() function returns a string object, while in Python 3 it returns a bytes object. 
Use six.text_type() to replace encode() to make code compatible both in Python2 and Python3. six.text_type() returns Unicode in Python2 and string in Python3.

2. Make sure string and bytes object are used correctly in translate().
Python 3 changes translate() to require a mapping table to be passed as an argument. Use ord() function to convert each character in the string '[]' to its corresponding Unicode code point, and then passes the resulting dictionary comprehension as an argument to the translate() method.

#### How did you verify/test it?
Run testcase manually.

### CODE REVIEW

**Comment by wsycqyz** [comment]
@nhe-NV Just let you know that I am actively working on the Python3 migration issues.

File: tests/qos/qos_sai_base.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import random
 import os
 import sys
+import six
```
Comment (neethajohn) [inline tests/qos/qos_sai_base.py:10]: There is already a open #7958

Comment (wsycqyz) [reply tests/qos/qos_sai_base.py:10]: Thanks for the information. I will try to fix conflict after #7958 is merged.

File: tests/qos/qos_sai_base.py [hunk 2]
```diff
@@ -240,9 +250,14 @@ def __getBufferProfile(self, request, dut_asic, os_version, table, port, priorit
...
-                argv=["redis-cli", "-n", db, "HGET", keystr, "profile"]
-            )[0].encode("utf-8").translate(None, "[]")
+            if six.PY2:
+                bufferProfileName = dut_asic.run_redis_cmd(
+                    argv=["redis-cli", "-n", db, "HGET", keystr, "profile"]
+                )[0].encode("utf-8").translate(None, "[]")
+            else:
+                bufferProfileName = dut_asic.run_redis_cmd(
+                    argv=["redis-cli", "-n", db, "HGET", keystr, "profile"]
+                )[0].translate({ord(i): None for i in '[]'})
```
Comment (wangxin) [inline tests/qos/qos_sai_base.py:260]: IMO, code like below has less duplication.
```
            out = dut_asic.run_redis_cmd(argv=["redis-cli", "-n", db, "HGET", keystr, "profile"])[0]
            if six.PY2:                
                bufferProfileName = out.encode("utf-8").translate(None, "[]")
            else:
                bufferProfileName = out.translate({ord(i): None for i in '[]'})
```

**Review by wangxin — APPROVED** [review]

**Pull request merged**